src/Microsoft.AspNet.Server.Kestrel/Http/SocketOutput.cs

```diff
@@ -48,6 +59,9 @@ public class SocketOutput : ISocketOutput
             _connectionId = connectionId;
             _log = log;
             _tasksPending = new Queue<TaskCompletionSource<object>>();
+
+            _head = memory.Lease();
```

Comment:
Yeah, would be nice if the SocketOutput could return itself to a `_head == _tail == null` state when there's known to be neither writes nor ProducingStart in effect at the time. Would help reduce memory pressure from long-running responses (web sockets or server-sent events) and reduce memory pressure for open connections waiting for the next request to arrive.